lib/resource_tools/json.rb

```diff
@@ -1,7 +1,46 @@
+# frozen_string_literal: true
+
+# Include in an ActiveRecord::Base object to handle JSON parsing and translation.
+#
+# @example Including a basic parser
+#   class MyClass < ApplicationRecord
+#     include ResourceTools
+#
+#     json do
+#       translate(
+#         side_walk: :pavement,
+#         trunk: :boot
+#       )
+#     end
+#   end
+#
+# The example above will create a new subclass of `ResourceTools::Json::Handler`
+# called `MyClass::JsonHandler`. This can be accessed via the class method
+# {ResourceTools::Json::json}
+#
+# The JsonHandler will automatically take parsed json (ie. covered to a Hash)
+# and will:
+# - convert each key to a string
+# - perform any translations specified via {ResourceTools::Json::Handler::translate}
+#
+# @example Invoking the Handler
+#   MyClass.create_or_update_from_json(parsed_json, 'Lims')
+#
+# This will extract the translated attributes from parsed_json and pass them
+# into the create_or_update method.
```

Comment:
Added this to the equivalent class in the Event Warehouse, so ported it across here.